File: 기초-브론즈 문제/백준_1773_폭죽쇼/폭죽쇼.java
import java.util.*;

public class Main{
    public static void main(String[] args){
        Scanner s = new Scanner(System.in);
        int n,c;
        n = s.nextInt();
        c = s.nextInt();
        int[] time = new int[n];
        boolean[] fire = new boolean[2000001];
        for(int i = 0; i< n; i++) time[i] = s.nextInt();

        int cnt = 0;
        for(int i = 0; i< n; i++){
            for(int j = time[i]; j <= c; j +=time[i]){
                if(fire[j]) continue;
                fire[j] = true;
                cnt++;
            }
        }
        System.out.println(cnt);
    }
}

File: 기초 사고력문제/백준_1773_폭죽쇼/폭죽쇼.kt
import java.util.*

fun main(args: Array<String>) {
    val s = Scanner(System.`in`)
    val n = s.nextInt()
    val c = s.nextInt()
    var fire = BooleanArray(2000001, {false})
    var time = IntArray(n, {0})
    for(i in 0..n-1)
        time[i] = s.nextInt()

    var cnt = 0
    for(i in 0..n-1){
        for(j in time[i]..c step time[i]){
            if(fire[j]) continue
            fire[j] = true
            cnt++
        }
    }
    print("${cnt}")

}


